optimum/bettertransformer/models/encoder_models.py

```diff
@@ -449,8 +462,11 @@ def forward(self, hidden_states, attention_mask, position_bias=None, *_, **__):
             self.linear2_bias,
             attention_mask,
         )
-        if hidden_states.is_nested and self.is_last_layer:
-            hidden_states = hidden_states.to_padded_tensor(0.0)
+
+        if not self.is_last_layer:
+            hidden_states.original_shape = original_shape
```

Comment:
we actually need to do it multiple times, to pass the shape from a layer to another up to the end; I think

Reply:
I really doubt this is needed to be passed on all layers since I don't excpect the hidden states shape to change over time, but it's a tiny complexity so we can leave it as it is I guess!

Reply:
Thing is that https://github.com/huggingface/optimum/blob/fb8e284c81a3e4818803d6a6fc8ee98e6c740cf1/optimum/bettertransformer/models/encoder_models.py#L431 **overwrites** the `hidden_states`, hence you need to rewrite `hidden_states.original_shape` at every layer. Do you agree?

Reply:
ahh yes! Great point